### PR TITLE
Copter: Obey and preserve DO_SET_ROI commands when executing LOITER_TURNS mission commands

### LINKED ISSUE #569: ArduCopter: make ROI persistent across waypoints
After testing firmware 3.1 rc2 it appears ROI is not functioning as it should/could.
1. According to the wiki it should be possible to influence yaw behaviour of DO_SET_ROI by passing a parameter in the mission planner waypoint list. Doing this has no effect on flying behaviour. Is this a bug?
2. It should be possible to set 1 ROI marker in mission planner that is used for orientation of a multicopter during the entire mission. Now it only influences the first waypoint following the DO_SET_ROI command. For the rest of the mission yaw is directed at the next waypoint. Is this by design or a bug?

### PR DESCRIPTION
**Pull request by amilcarlucas**
Fixes #569

### CODE REVIEW

**Comment by rmackay9** [comment]
@amilcarlucas,
It looks mostly ok, but I wonder, is the purpose:

#. to allow pointing at a point other than the center of the circle
#. to point at the center of the circle more accurately (the current circle control tends to point slightly behind the center of the circle.

If it's (2) then I think we should fix the problem's root cause instead of providing a work around.  If it's (1) then the change is fine I think.

**Comment by amilcarlucas** [comment]
Hi Randy,

Yes, the point id to provide the ability to obey DO_SET_ROI in LOITER_TURNS commands. So 1. on your list.
It also provides the ability to not mess up DO_SET_ROI: only a single DO_SET_ROI is required then you can use any combination of WAYPOINTS and LOITER_TURNS commands and it will be preserved throughout them.

Even with this method I do see that with some circle radius, at some radial speeds it tends to point "behind" the center for bearings in [0; 180] and "in front of" the center for bearings in [180; 360].
Basically I get in some configurations a pointing error as a sinus function of the bearing with an amplitude of up to 5 degrees. But that is a different bug, requiring a different patch. I'll open another pull request if I'm able to find it and fix it.

**Comment by lvale** [comment]
@rmackay9 I believe that as it is, ie your item 1 is a valuable feature, and what might be most wanted by the majority of users. It'll depend a bit on how GCS's represent this, and how the wiki is written.

**Comment by OXINARF** [comment]
> If we're going to make this change, it probably makes sense that ROI is active (until turned off) for all navigation commands.

I think this is how it should work, it respects what the user asked for.

**Comment by amilcarlucas** [comment]
Looks like @lvale and @OXINARF approve this bugfix.
@rmackay9 should I rebase this or close it ?

**Comment by amilcarlucas** [comment]
Rebased

**Comment by lvale** [comment]
@rmackay9 Can you merge this before 3.6? Thanks.

**Comment by amilcarlucas** [comment]
ping

**Comment by amilcarlucas** [comment]
ping

**Comment by rmackay9** [comment]
@amilcarlucas, can you fix the commit to start with "Copter:"?  Also there's some issues with the comments and a spacing issue before one of the "true"s.  After that I'll merge it in, thanks!

**Comment by amilcarlucas** [comment]
Done

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by amilcarlucas** [comment]
@rmackay9: @lvale asked if this can be backported to 3.6

**Comment by lvale** [comment]
thanks @amilcarlucas 

That was 4 months ago, guess now would only delay the release of 3.6
